### ISSUE TITLE
Unable to build interface on Spaces

### ISSUE THREAD

**Issue by merveenoyan** [gradio-app/gradio#2095]
### Describe the bug

Hello 👋 
I have an interface (will post the code and the trace below) where I can run on my local (noted that my gradio version and other things are the same in the Space and the local). I get 
```
  File "/home/user/.local/lib/python3.8/site-packages/gradio/components.py", line 2660, in preprocess
    if x.get("headers") is not None:
AttributeError: 'list' object has no attribute 'get'
```
Link to Space: https://huggingface.co/spaces/scikit-learn/tabular-playground

It's a bit weird because on my local everything works fine and I have the same gradio version. I wonder if it's a Spaces problem or a gradio one but thought you might have more idea given the trace.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

```python
import sklearn
import gradio as gr
import joblib
import pandas as pd

pipe = joblib.load("./model.pkl")

title = "Supersoaker Defective Product Prediction"
description = "This model predicts Supersoaker production line failures."


with open("./config.json") as f:
    config_dict = eval(f.read())
headers = config_dict["sklearn"]["columns"]

example_dict = config_dict["sklearn"]["example_input"]
df = pd.DataFrame.from_dict(example_dict,orient='index').transpose()
examples=df.to_numpy().tolist()
final_examples = [[[example]] for example in examples]

inputs = gr.Dataframe(headers = [item for item in example_dict])
outputs = gr.Dataframe(headers = ["results"])


def infer(inputs):
    data = pd.DataFrame(inputs, columns=[item for item in example_dict])
    predictions = pipe.predict(inputs)
    return pd.DataFrame(predictions, columns=["results"])

gr.Interface(infer, inputs = inputs, outputs = outputs, title = title,
            description = description, examples=final_examples).launch(debug=True)


```

### Logs

```shell
/home/user/.local/lib/python3.8/site-packages/gradio/interface.py:318: UserWarning: Currently, only the 'default' theme is supported.
  warnings.warn("Currently, only the 'default' theme is supported.")
Caching examples at: '/home/user/app/gradio_cached_examples/14/log.csv'
Traceback (most recent call last):
  File "app.py", line 30, in <module>
    gr.Interface(infer, inputs = inputs, outputs = outputs, title = title,
  File "/home/user/.local/lib/python3.8/site-packages/gradio/interface.py", line 603, in __init__
    self.examples_handler = Examples(
  File "/home/user/.local/lib/python3.8/site-packages/gradio/examples.py", line 52, in create_examples
    utils.synchronize_async(examples_obj.create)
  File "/home/user/.local/lib/python3.8/site-packages/gradio/utils.py", line 353, in synchronize_async
    return fsspec.asyn.sync(fsspec.asyn.get_loop(), func, *args, **kwargs)
  File "/home/user/.local/lib/python3.8/site-packages/fsspec/asyn.py", line 96, in sync
    raise return_result
  File "/home/user/.local/lib/python3.8/site-packages/fsspec/asyn.py", line 53, in _runner
    result[0] = await coro
  File "/home/user/.local/lib/python3.8/site-packages/gradio/examples.py", line 192, in create
    await self.cache_interface_examples()
  File "/home/user/.local/lib/python3.8/site-packages/gradio/examples.py", line 229, in cache_interface_examples
    raise e
  File "/home/user/.local/lib/python3.8/site-packages/gradio/examples.py", line 225, in cache_interface_examples
    prediction = await self.process_example(example_id)
  File "/home/user/.local/lib/python3.8/site-packages/gradio/examples.py", line 241, in process_example
    processed_input = [
  File "/home/user/.local/lib/python3.8/site-packages/gradio/examples.py", line 242, in <listcomp>
    input_component.preprocess(raw_input[i])
  File "/home/user/.local/lib/python3.8/site-packages/gradio/components.py", line 2660, in preprocess
    if x.get("headers") is not None:
AttributeError: 'list' object has no attribute 'get'
```

### System Info

```shell
gradio==3.1.7
```

### Severity

annoying

**Comment by abidlabs** [comment]
Hey @merveenoyan it looks like the issue is happening because of example caching (which happens by default in Spaces, but not elsewhere). You set `cache_examples=False` in `Interface()` to prevent this from happening in Spaces, while we look into the underlying issue. Thanks!

**Comment by abidlabs** [comment]
Following up here, I rebuilt the same Space using `gradio==3.1.8b0`, and the Space seems to build correctly: https://huggingface.co/spaces/abidlabs/tabular-playground/tree/main

So this issue should be fixed when we release 3.1.8 next week. (Besides the examples not showing up properly, which we'll tackle as part of #2092)

**Comment by merveenoyan** [comment]
it works now, thanks! https://huggingface.co/spaces/scikit-learn/tabular-playground
